sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/ChatRequestAssistantMessage.java

```diff
@@ -181,7 +187,17 @@ public static ChatRequestAssistantMessage fromJson(JsonReader jsonReader) throws
                 String fieldName = reader.getFieldName();
                 reader.nextToken();
                 if ("content".equals(fieldName)) {
-                    content = reader.getNullable(nonNullReader -> BinaryData.fromObject(nonNullReader.readUntyped()));
+                    if (reader.currentToken() == JsonToken.STRING) {
+                        content = BinaryData.fromString(reader.getString());
+                    } else if (reader.currentToken() == JsonToken.START_ARRAY) {
+                        content = BinaryData.fromObject(
+                            reader.readArray(arrayReader -> arrayReader.readObject(ChatMessageContentItem::fromJson)));
+                    } else if (reader.currentToken() == JsonToken.NULL) {
+                        content = null;
+                    } else {
+                        throw new IllegalStateException("Unexpected 'content' type found when deserializing"
+                            + " ChatRequestAssistantMessage JSON object: " + reader.currentToken());
+                    }
```

Comment:
I think we may want to be less strict here, specially if at some point we don't have parity with non-Azure OpenAI. WDYT? Also, this might be a non issue, since this class seems to be used only for request and will therefore never be serialized.

Reply:
I would prefer to be strict as we know other token types should not be serialized. I would leave it as it is for now and we can update it later when "parity with Non-Azure OpenAI" raised.